### PR TITLE
test: Detect misconfigured coredump handling

### PR DESCRIPTION
**Pull request by dzafman**
## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dzafman** [comment]
@liewegas Let's detect systemd-cordump active and fail, so we will find this immediately if it happens again.

**Comment by dzafman** [comment]
jenkins test make check

**Comment by dzafman** [comment]
jenkins test make check

File: qa/standalone/ceph-helpers.sh [hunk 1]
```diff
@@ -172,12 +172,10 @@ function teardown() {
     local pattern="$(sysctl -n $KERNCORE)"
     # See if we have apport core handling
     if [ "${pattern:0:1}" = "|" ]; then
-      # TODO: Where can we get the dumps?
-      # Not sure where the dumps really are so this will look in the CWD
-      pattern=""
-    fi
+      echo "Misconfigured test environment, systemd-coredump not supported"
```
Comment (tchaikov) [inline qa/standalone/ceph-helpers.sh]: i logged on to a random smithi xenial test node runing test. and ran
```console
kchai@smithi109:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
$ sysctl -n kernel.core_pattern
/home/ubuntu/cephtest/archive/coredump/%t.%p.core
```
so it seems that `kernel.core_pattern` was set as expected.

**Comment by liewegas** [comment]
I think this bug was fixed by https://github.com/ceph/teuthology/pull/1395 ... one of the package installs was upgrading systemd and restarting the service, which reset the sysctl value.

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
this makes jenkins 'make check' fail:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/standalone/ceph-helpers.sh:175: teardown:  echo 'Misconfigured test environment, systemd-coredump not supported'
Misconfigured test environment, systemd-coredump not supported
```

**Comment by dzafman** [comment]
Not worth the time to special case the make check tests to bypass this check.

**Pull request closed without merging**